core/src/main/java/com/linecorp/armeria/internal/common/RequestContextUtil.java

```diff
@@ -225,20 +225,25 @@ public static void pop(RequestContext current, @Nullable RequestContext toRestor
     public static SafeCloseable invokeHookAndPop(RequestContext current, @Nullable RequestContext toRestore) {
         requireNonNull(current, "current");
 
-        final SafeCloseable closeable = invokeHook(current);
+        final AutoCloseable closeable = invokeHook(current);
         if (closeable == null) {
             return () -> requestContextStorage.pop(current, toRestore);
         } else {
             return () -> {
-                closeable.close();
+                try {
+                    closeable.close();
+                } catch (Throwable t) {
+                    logger.warn("Unexpected exception while closing RequestContext.hook(). " +
+                                "closeable: {}, ctx: {}", closeable, current, t);
```

Comment:
We always put ctx in the beginning of the log message.

```
"{} Unexpected ...", ctx, ...
```

Is stringifying `closeable` useful? Drop it if not.